### PR TITLE
Add document for etcd state store

### PR DESCRIPTION
**Pull request by taenyang**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [x] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [x] Commands include options for Linux, MacOS, and Windows within codetabs
- [x] New file and folder names are globally unique
- [x] Page references use shortcodes instead of markdown or URL links
- [x] Images use HTML style and have alternative text
- [x] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

Add document to describe how to use etcd state store

## Issue reference

[#3270](https://github.com/dapr/docs/issues/3270)
[#3269](https://github.com/dapr/docs/issues/3269)

### CODE REVIEW

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-etcd.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      - "12379:2379"
+    command: etcd --listen-client-urls http://0.0.0.0:2379 --advertise-client-urls http://0.0.0.0:2379```
+```
+You can then interact with the server using `localhost:12379`.
+
+{{% /codetab %}}
+
+{{% codetab %}}
+
+We can use [Helm](https://helm.sh/) to quickly create an Etcd instance in our Kubernetes cluster. This approach requires [Installing Helm](https://github.com/helm/helm#install).
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-etcd.md]:
```suggestion
Use [Helm](https://helm.sh/) to quickly create an Etcd instance in your Kubernetes cluster. This approach requires [Installing Helm](https://github.com/helm/helm#install).
```

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-etcd.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+Follow the instructions [here](https://github.com/bitnami/charts/tree/main/bitnami/etcd) to get started with setting up Etcd in Kubernetes.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-etcd.md]:
```suggestion
Follow the [Bitnami instructions](https://github.com/bitnami/charts/tree/main/bitnami/etcd) to get started with setting up Etcd in Kubernetes.
```

**Review by hhunter-ms — CHANGES_REQUESTED** [review]

**Review by hhunter-ms — APPROVED** [review]

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-etcd.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+{{% /alert %}}
+
+## Spec metadata fields
+
+| Field              | Required | Details | Example |
+|--------------------|:--------:|---------|---------|
+| endpoints          | Y        | Connection string to Etcd server | `"192.168.0.1:2379,192.168.0.2:2379,192.168.0.3:2379"`
+| keyPrefixPath      | N        | Key prefix path in Etcd. Default is `""` | `"dapr"`
+| tlsEnable          | N        | Whether to enable tls  | `"false"`
+| ca                 | N        | Contents of Etcd server CA file | `"ca value"`
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-etcd.md]: @taenyang - For the `ca`, `cert` and `key` examples we have found that either point to documentation or providing more real world examples are better here. The goal here is clarity if there are any specific on how these value should be provided i.e. the Format of the strings. Is there anything special needed?

Comment (taenyang) [reply daprdocs/content/en/reference/components-reference/supported-state-stores/setup-etcd.md]: Add reference to use component-secrets and real example values for ca, cert and key.

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-etcd.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+| cert               | N        | Contents of Etcd server certificate file. Can be `secretKeyRef` to use a [secret reference]({{< ref component-secrets.md >}}).| `"-----BEGIN CERTIFICATE-----\nMIIDUTCC..."`
+| key                | N        | Contents of Etcd server key file. Can be `secretKeyRef` to use a [secret reference]({{< ref component-secrets.md >}}).| `"-----BEGIN RSA PRIVATE KEY-----\nMIIEpAIB..."`
+
+## Setup Etcd
+
+{{< tabs "Self-Hosted" "Kubernetes" >}}
+
+{{% codetab %}}
+
+You can run Etcd locally using Docker Compose:
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-etcd.md]:
```suggestion
You can run Etcd database locally using Docker Compose. Create a new file called `docker-compose.yml` and add the following contents as an example:
```

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-etcd.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+```
+version: '2'
+services:
+  etcd:
+    image: gcr.io/etcd-development/etcd:v3.4.20
+    ports:
+      - "12379:2379"
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-etcd.md]:
```suggestion
      - "2379:2379"
```

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-etcd.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    command: etcd --listen-client-urls http://0.0.0.0:2379 --advertise-client-urls http://0.0.0.0:2379```
+```
+You can then interact with the server using `localhost:12379`.
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-etcd.md]:
```suggestion
Save the docker-compose.yml file and run the following command to start the Etcd server:

`docker-compose up -d`

This starts the Etcd server in the background using the expose the default Etcd port of `2379`. You can then interact with the server using the `etcdctl` command-line client on `localhost:12379`. For example

```
etcdctl --endpoints=localhost:2379 put mykey myvalue

```

```

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-etcd.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+    ports:
+      - "12379:2379"
+    command: etcd --listen-client-urls http://0.0.0.0:2379 --advertise-client-urls http://0.0.0.0:2379```
+```
+Save the docker-compose.yml file and run the following command to start the Etcd server:
+
+`docker-compose up -d`
+
+This starts the Etcd server in the background using the expose the default Etcd port of `2379`. You can then interact with the server using the `etcdctl` command-line client on `localhost:12379`. For example
+
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-etcd.md:77]:
```
etcdctl --endpoints=localhost:2379 put mykey myvalue
```

**Pull request merged**